### PR TITLE
DriverBehaviour added

### PR DESCRIPTION
**Pull request by plitvica**
attributes, setter, getter added

### CODE REVIEW

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 1]
```diff
@@ -3,4 +3,41 @@
 import at.fhv.itm3.s2.roundabout.api.entity.IDriverBehaviour;
 
 public class DriverBehaviour implements IDriverBehaviour {
+
+    private double speed;
+    private double minDistanceToNextCar;
+    private double maxDistanceToNextCar;
+    private double behaviourFactor;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: mergeFactor in entity diagram, whats the purpose of this value?

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 2]
```diff
@@ -3,4 +3,41 @@
...
+
+    public double getSpeed() {
+        return speed;
+    }
+
+    public void setSpeed(double speed) {
+        this.speed = speed;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: checks, for example if positive

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 3]
```diff
@@ -3,4 +3,41 @@
...
+    }
+
+    public double getMinDistanceToNextCar() {
+        return minDistanceToNextCar;
+    }
+
+    public void setMinDistanceToNextCar(double minDistanceToNextCar) {
+        this.minDistanceToNextCar = minDistanceToNextCar;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: checks, for example if positive

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 4]
```diff
@@ -3,4 +3,41 @@
...
+    }
+
+    public double getMaxDistanceToNextCar() {
+        return maxDistanceToNextCar;
+    }
+
+    public void setMaxDistanceToNextCar(double maxDistanceToNextCar) {
+        this.maxDistanceToNextCar = maxDistanceToNextCar;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: checks, for example if positive

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 5]
```diff
@@ -3,4 +3,41 @@
...
+    }
+
+    public double getBehaviourFactor() {
+        return behaviourFactor;
+    }
+
+    public void setBehaviourFactor(double behaviourFactor) {
+        this.behaviourFactor = behaviourFactor;
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: checks, for example if positive or in specific range

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java (hunk 1 above)
Comment (Oxydation) [reply roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: also, to be within our coding guidelines, i would advise to use the _ as prefix of a private variable, like _speed;

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java (hunk 5 above)
Comment (Oxydation) [reply roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: i also think we have renamed this one to something like "mergeFactor" or so.. take some time to overthink the naming of this one

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java (hunk 1 above)
Comment (DaGrisa) [reply roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: no, this is bad practice: https://google.github.io/styleguide/javaguide.html

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java (hunk 5 above)
Comment (DaGrisa) [reply roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: like mentioned in the comment of the attribute...

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 6]
```diff
@@ -3,4 +3,41 @@
 import at.fhv.itm3.s2.roundabout.api.entity.IDriverBehaviour;
 
 public class DriverBehaviour implements IDriverBehaviour {
+
+    private double speed;
+    private double minDistanceToNextCar;
+    private double maxDistanceToNextCar;
+    private double behaviourFactor;
+
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: maybe also create a constructor with all attributes

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 7]
```diff
@@ -3,4 +3,58 @@
...
+        this.mergeFactor = mergeFactor;
+    }
+
+
+    public double getSpeed() {
+        return speed;
+    }
+
+    public void setSpeed(double speed) {
+        if(speed > 0){
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: ```>=``` is better

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 8]
```diff
@@ -3,4 +3,58 @@
...
+            this.speed = speed;
+        }
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: else throw exception

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 9]
```diff
@@ -3,4 +3,58 @@
...
+    }
+
+    public double getMinDistanceToNextCar() {
+        return minDistanceToNextCar;
+    }
+
+    public void setMinDistanceToNextCar(double minDistanceToNextCar) {
+        if(minDistanceToNextCar > 0){
+            this.minDistanceToNextCar = minDistanceToNextCar;
+        }
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: else throw exception

File: roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java [hunk 10]
```diff
@@ -3,4 +3,58 @@
...
+    }
+
+    public double getMaxDistanceToNextCar() {
+        return maxDistanceToNextCar;
+    }
+
+    public void setMaxDistanceToNextCar(double maxDistanceToNextCar) {
+        if(maxDistanceToNextCar > 0){
+            this.maxDistanceToNextCar = maxDistanceToNextCar;
+        }
```
Comment (DaGrisa) [inline roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/DriverBehaviour.java]: else throw exception

**Pull request merged**